### ISSUE TITLE
[HELP WANTED] Perform source-map audit against webpack

### ISSUE THREAD

**Issue by TheLarkInn** [webpack/webpack#3165]
## Problem

There are multiple issues that have been filed for numerous different reports of source-maps having issues in chrome. 
## Work Required

Listed is the canonical specification for source-maps:
https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k/edit

We would like help (as we cannot make this a priority with current resources) creating every type of sourcemap, and then either manually (or via unit testing) validate that the source-map types that we have configured in place currently (starting at webpack2.1.0-beta.25) and work in Chrome. 
## Results

This will allow us to:
- Create unit tests that are sturdier, and a source map validation tool to ensure it works to this specification (if it doesn't already exist). 
- Identify if source-map issues are chrome bugs or webpack bugs.

If there is anyone who would like to tackle this, please comment in the issue, then we can be aware of who is working on it, and how

**Comment by addyosmani** [comment]
Have you looked at https://github.com/mattrobenolt/sourcemap-validator before? I believe in the past they've tried to roughly approximate how well support with the current spec could be captured by their validator.

**Comment by mattkime** [comment]
I made a small (very small) hello world script and passed it through webpack. the resulting sourcemap does not work in chrome. 

https://github.com/mattkime/node-test-debuggers

`npm install && npm run build && npm run chrome:webpack` and you should be able to see the issue - the original 'index.js' file is referenced in the sourcemap but its not displayed in the sources list.

**Comment by TheLarkInn** [comment]
@addyosmani thank you so much. 

@mattkime are you familiar will all the types of sourcemaps you can generate with webpack?

**Comment by TheLarkInn** [comment]
See https://github.com/webpack/webpack/blob/master/lib/WebpackOptionsApply.js#L200-L232

**Comment by mattkime** [comment]
@TheLarkInn I am. I guess I should iterate through all the options? I was trying to find the simplest starting point and iterate out through there.

**Comment by TheLarkInn** [comment]
Yeah id say just one at a time! I haven't tried out the validator that Addy suggested just yet but I would think that could provide a great way to iterate quickly.

**Comment by mattkime** [comment]
The validator is, um, under documented. I gave 'eval-soucre-map' a go and it worked well on my simple test and failed on a complex project.

As mention in the issue description, I'm not sure if i'm looking at webpack or chrome bugs. 

While I'm happy to pitch in, I need some guidance to get started on this.

**Comment by mattkime** [comment]
Ideally we'd find someone on the Google Chrome project that could give us some guidance.

**Comment by joachimprinzbach** [comment]
I did also like this link for some information: http://jlongster.com/On-the-Road-to-Better-Sourcemaps-in-the-Firefox-Developer-Tools

**Comment by mattkime** [comment]
this shows a sourcemap working when loaded via front end and failing when loaded via node
https://github.com/mattkime/chrome-devtools-sourcemap-test

**Comment by rob3c** [comment]
Sorry I don't have anything productive to add yet beyond an observation, but maybe now that angular 2 is promoting typescript and their angular-cli project is depending on webpack 2, it's possible someone on the angular team will have a better-than-average chance of getting some feedback from the chrome team on this? As angular 2 picks up users, I have to imagine that'll mean a lot of new users wondering why sourcemaps are broken.
https://github.com/angular/angular-cli

**Comment by mattkime** [comment]
verified the problem on chromium and opened a bug. it would be helpful if people could verify the bug on windows and linux. https://bugs.chromium.org/p/chromium/issues/detail?id=658438

would probably help to get a bunch of followers on that bug as well.

**Comment by TheLarkInn** [comment]
@rob3c I'm working with the team to get some help and feedback. cc @PaulIrish

**Comment by TheLarkInn** [comment]
@mattkime have you run your test against each sourcemap type. Should we make a chart or checklist?

**Comment by mattkime** [comment]
@TheLarkInn the bug i submitted to chromium uses babel rather than webpack. the output is a bit simpler but still exposes the flaw.

lets make a new project that makes it easy to iterate over the various sourcemap types, in a similar spirit as https://github.com/mattkime/node-test-debuggers. 

i did try a couple of other sourcemap types and found that they may work in a simple example and then fail in a complex setup. My simple test is not sufficient to determine whether a given sourcemap type works in completion.

**Comment by mattkime** [comment]
@TheLarkInn v1.13, right?

**Comment by mattkime** [comment]
threw this together, might be rough around the edges - https://github.com/mattkime/webpack-sourcemap-test

prompts you for the devtool you'd like to build with and then creates node and web builds.

**Comment by TheLarkInn** [comment]
Curious about v2.1.0-beta.25 also

**Comment by TheLarkInn** [comment]
Nice work btw. I will check this out.

**Comment by mattkime** [comment]
>>Curious about v2.1.0-beta.25 also

Haven't double checked but i suspect the problem would be reproduced identically.

`eval` devtool seems to work but i suspect thats because its not really a sourcemap.

**Comment by jmfirth** [comment]
The best, working option I have found so far is `eval-source-map`.

**Comment by mattkime** [comment]
@jmfirth which OS, node version, and chrome version are you using?

i just tried `eval-source-map` and did not see my source files

**Comment by jmfirth** [comment]
Apologies.  `eval-source-map` worked for me on Win 10 with node 6.9.0, webpack 1.12.13 and Chrome 53.0.2785.143.

**Comment by TheLarkInn** [comment]
Since this is going to be a webpack 2+ addition, I would like have all verifications being done against the latest version of webpack (HEAD/webpack@2.1.0-beta.25).

**Comment by mattkime** [comment]
@jmfirth is that project closed source or open source?

**Comment by jmfirth** [comment]
@mattkime Closed (business app)..  The above also works with Chrome 54.0.2840.71.

@TheLarkInn makes sense.  Was hoping to help a person like me who stumbles across these recent issues with Chrome in configuring a lesser but working environment.

**Comment by mattkime** [comment]
@jmfirth tried that version of webpack, node, and chrome but it didn't work. I'd like to try on a windows machine. would be nice if you could try on a mac.

**Comment by octref** [comment]
People in this thread probably know much more than an average dev about sourcemaps. But none seem to know exactly which type works with which environment (some of them don't work together).

Now imagine users' frustration trying to make webpack work with this:

![image](https://cloud.githubusercontent.com/assets/4033249/20032266/dc51c2ae-a343-11e6-8f56-3e1e3101fcc7.png)

It's great to see this problem is being actively addressed. But as an end user, I hope more efforts can be made to:

- Have a type of sourcemap that "just works" across environments.
- Make it default. Like `sourcemap: true` for the default and If people want more advanced config let them provide `devtool`.

And, what is `SourceMapDevToolPlugin`? I've seen people using it with webpack2, but there is no doc for it. Is it preferred over `devtool`? Why is it not audited here?

Finally, @mattkime if you are familiar with all these types of sourcemap, it'll be awesome to see you writing some doc (or a blog post) that explains what each of them does, in plain terms. You'll be doing a great favor to all webpack users like me and [this guy who just opened an issue on webpack](https://github.com/webpack/webpack/issues/3240).

**Comment by TheLarkInn** [comment]
@octref the plugins are just what webpack does behind the scenes for the devtool prop on the config.  Each map type has a trade-off as seen from the table. I really want to ensure that we are:

1. Aligning with spec, if so, and if still issues, report to @paulirish and friends and report to chromium.
2. If not aligning with spec, fix bugs in our code.

**Comment by bennewton999** [comment]
Setting up a simple demo from an online tutorial using Webpack 2 - source map is not working in Chrome, but is working in Canary.

config is:
devtool: 'source-map'

console.log in app.js shows in console as:

Chrome: bundle.js Line 1
Canary: app.js Line 3

Versions:
Webpack: 2.1.0-beta.27
Chrome : 54.0.2840.98 (64-bit)
Canary: 57.0.2931.0 canary (64-bit)

**Comment by octref** [comment]
@TheLarkInn 

My point is webpack fails to be user friendly by requiring user reading some arcane spec to figure out which sourcemap config to use.

**Comment by TheLarkInn** [comment]
@octref 

> My point is webpack fails to be user friendly by requiring user reading some arcane spec to figure out which sourcemap config to use.

I'm sorry if I didn't make anything clear. The purpose of the issue is so we can make sure each type of sourcemap we provide works. Sourcemaps can be slow, and so we created a variety of solutions/type of sourcemap based, so you can have more flexibility. Thus the table you screenshot'd above.

Overall, I believe that all of the versions should "just work" no matter if you are using a default from the -d or -p flag from the cli, or `devtool` property. The spec is just a tool for anyone who wants to help contribute to our efforts to ensure we are creating and building them correctly, and determine if it is our fault or Chromes fault. Hope that makes sense. Let me know if I can help more.

**Comment by octref** [comment]
@TheLarkInn Thanks. Glad to hear the goal is to make all versions "just work". Hope that goal is achieved someday.

**Comment by TheLarkInn** [comment]
Likewise! I _wish_ we could make it the number one priority however we are very close to releasing an RC for v2, therefore created this in hopes to have assistance from the community until we swing back around to it.

**Comment by chemzqm** [comment]
I have a module https://github.com/chemzqm/stack-source-map that makes Chrome error stack works with every kind of sourcemap options, one problem is I have to use synchronize XHR request to get the source code.

 I believe that most issue belongs to Chrome,  the `cheap-module-eval-source-map` no longer works after I upgrade Chrome.

**Comment by TheLarkInn** [comment]
Interesting. Thank you for this.

**Comment by caesarsol** [comment]
Hello all,
I'd be interested in the status of this issue, since the final 2.2 is out and the [devtool documentation](https://webpack.js.org/configuration/devtool/) links to this issue.

I always found the multiple devtool options (in webpack 1) to be very confusing, I'd like to help by rewriting that documentation page!

**Comment by Vanuan** [comment]
Here's my feedback:

Webpack version: `2.2.0`
Babel config:
```
  presets: [
    'react',
    ['es2015', { 'modules': false }],
    'stage-1'
    ],
    plugins: ['transform-decorators-legacy']
  ]
```
Devtool options:

1. cheap-module-source-map

Breakpoints work but are mapped to the wrong place. Offset is approximately 3 lines for each import statement.
Bugs in import are not debuggable (e.g. `"__webpack_require__.i(...)" is not a function`)

2. eval

Breakpoints work and are mapped to the correct place. Bugs in imports are debuggable.
Problems:
- You can't set breakpoints in the compiled file: it uses eval('string'), you can only set them in sources.
- `__webpack_require__.i` instead of module names

3. eval-source-map

Looks the same as cheap-module-source-map

4. cheap-source-map

The same as cheap-module-source-map


So, it looks like all the options except "eval" (generated code) are broken.

**Comment by paulirish** [comment]
It'd be useful to load some of those examples into http://paulirish.github.io/source-map-visualization/ to see if it also reflects the same offsets.

**Comment by fracmak** [comment]
I loaded my example into the visualizer and it's definitely wrong in the visualizer as well. Here's the files I uploaded.

[bundle.zip](https://github.com/webpack/webpack/files/784057/bundle.zip)

**Comment by Vanuan** [comment]
cheap-module-source-map:

![screenshot from 2017-02-17 21 03 24](https://cloud.githubusercontent.com/assets/361985/23079228/a7f56ec6-f554-11e6-8d52-2916f6e0d5e7.png)

**Comment by fracmak** [comment]
<img width="1338" alt="screenshot 2017-02-17 14 08 56" src="https://cloud.githubusercontent.com/assets/378178/23079365/ad875a0a-f51a-11e6-84e6-b91d859fba96.png">

**Comment by Vanuan** [comment]
So it appears that this line
```
Object.defineProperty(__webpack_exports__, "__esModule", { value: true });
```
is not taken into an account.

**Comment by TheLarkInn** [comment]
@vanuan @paulirish thank you so much for this research and feedback. It is greatly appreciated. I think I've found where this is occurring and may be a webpack related bug. (I'll try and find the issue).

**Comment by chemzqm** [comment]
https://www.npmjs.com/package/concat-with-sourcemaps could helps for concat sourcemaps.

**Comment by TheLarkInn** [comment]
https://github.com/webpack/webpack/issues/3579 going to tie this here as this is actively in our milestone to be fixed. It appears that there is an issue in http://github.com/webpack/source-list-map where new lines are not being taken into consideration correctly. I'm going to try to tackle this but I won't be offended if someone beats me to the chase.

**Comment by Vanuan** [comment]
Any update?

**Comment by rbosneag** [comment]
@TheLarkInn actually, the line numbers are incorrect even if I disable `devTool`, the entire _app.js_ bundle has an offset of one line in Chrome

**Comment by TheLarkInn** [comment]
Yes there is a bug in github.com/webpack/source-list-map I've been trying to hunt down where return lines from single to multi are not being handled correctly.

**Comment by rbosneag** [comment]
you mean it still uses _github.com/webpack/source-list-map_ even if no source maps are generated? I disabled `devtool` and still have a line drift

**Comment by Tiagojdferreira** [comment]
@TheLarkInn any news regarding this issue?

**Comment by TheLarkInn** [comment]
Hey all thanks for the interest in this ongoing. For a good synopsis of the research done last night you can see https://t.co/eFp1Q0D9y0.

So we believe the the issue is specific to using Harmony(es6 imports/exports). Essentially there is somewhere between webpack/webpack-sources, webpack/source-list-map, webpack/webpack where a new line return is not being handled correctly. 

This has been hard to track down because of the nature of how es6 imports map to webpack code. For example a single HamonyImportSpecifier may be one line in userland code, however outputs as multiple lines. 

 Also a case in vice versa can occur where:
```
import {bar} from './foo';
import {baz} from '.'foo'; 
```

Will yield one line of output code.

**Comment by Vanuan** [comment]
I think I figured out why breakpoints aren't working with `eval` option:

Since the script you added breakpoint to is added using `eval()`, the line of code is executed before devtool knows about its existence.

I think this should be added to documentation.

**Comment by wbern** [comment]
### Work-around for Chrome found!
`eval` and `source-map` options work for me in Chrome Beta, version 58. The remaining options are still broken in various ways described below.

**Version 56.0.2924.87 (64-bit)**
* Nothing works. Filenames are slightly off with some extra letters on each of them. Sometimes it works sometimes it doesn't, even using `source-map`

**Version 58.0.3029.19 _beta_ (64-bit)**
* source-map: check
* eval-source-map: check, with suspicious text at end of file-name. See image.
![image](https://cloud.githubusercontent.com/assets/25477382/24100071/b72e8f90-0d72-11e7-9231-5b8adb4ebec7.png)
* cheap-module-source-map & cheap-module-eval-source-map & cheap-source-map: 
  -> debugger; statements don't seem to cause breakpoints to appear correctly in the devtool
  -> placing breakpoints feels off and they don't trigger either.
  -> weird text at end of filename still present.
![image](https://cloud.githubusercontent.com/assets/25477382/24100182/4485d81c-0d73-11e7-95ba-c91fa4a24261.png)
* nosources-source-map: everything is just blank. filenames look OK though.
![image](https://cloud.githubusercontent.com/assets/25477382/24100413/6563b2ce-0d74-11e7-9ce2-eaefbcb8b10c.png)
* eval: check, seems to work! Looks a bit weird in places, but maybe that's just the nature of using eval and react together (createClass functions here and there for example).
![image](https://cloud.githubusercontent.com/assets/25477382/24100496/cc3e227c-0d74-11e7-893e-e541d19254f2.png)

Hope this helps @TheLarkInn

**Comment by wbern** [comment]
Hmm, I may have to go back on what I said before, because it doesn't work reliably after all, after your first edit and the webpack dev server re-compiles.

![image](https://cloud.githubusercontent.com/assets/25477382/24139451/645e6826-0e1c-11e7-822b-bd3fe68d983f.png)

* Run `webpack-dev-server` with `source-map` on
* Make a breakpoint in the bundle, devtool will automatically switch you to the view of that particular file the code line belongs to (file.js for example)
* Modify some of the code in your usual editor, let webpack recompile
* Refresh the page, and try to place a breakpoint again. It will be broken at this point.

**Comment by wbern** [comment]
I noticed that 2 source maps are being generated, so I looked at the source map file in the browser before and after modifying the code.

The original source map is 4189930 characters, and after the change it's just a measly 9049, suggesting that the source map now suddenly only reflects the latest changes??

**Comment by wbern** [comment]
Ignore my previous posts, I had a configuration issue. My main issue that I stated below is because I am overriding the sourcemap filename in the webpack config. I should just remove it (aka let it be `[file].map` which is the default value). When I was specifically setting the source map filename, it was overriding any previous source maps, causing stuff to break upon page refresh.

Good lesson to anyone who reads, if you're using HMR / React hot loader, make sure you don't override source map file names for no good reason.

**Comment by Vanuan** [comment]
Found 2 issues:

1. Harmony imports: https://github.com/webpack/webpack-sources/issues/16
2. Destructuring: https://github.com/webpack/webpack-sources/issues/15

**Comment by Tiagojdferreira** [comment]
Any news ?

I know this is a hard one and I am just glad you guys are working so hard to fix it.  😃

**Comment by Vanuan** [comment]
@Tiagojdferreira `source-map` configuration without uglify should be good in webpack@2.3.2

**Comment by ldstein** [comment]
Hi, I think the content-type sent by the server for a sourcemap might be affecting the alignment in Chrome Debugger. 

- Issue logged: webpack/webpack-dev-server#891
- How I tested: https://github.com/ldstein/webpack-sourcemap-test

**Comment by lanoxx** [comment]
I was being hit by this bug earlier today. After searching for a while I found this [documentation](https://developers.google.com/web/tools/chrome-devtools/javascript/source-maps):

> TL;DR
>     [...]
>     Verify that your web server can serve Source Maps. 

I was just using IntelliJ's built in WebServer which was serving on port 63342. Chrome showed me the noticed about "Source maps detected" but did not load the source maps. I then installed `webpack-dev-server` and now there is a `webpack://`that contains the mapped source code.

**Comment by bbottema** [comment]
@Vanuan 

> @Tiagojdferreira source-map configuration without uglify should be good in webpack@2.3.2
>
>Other configurations have minor issues.

Not in Chrome 58.0.3029.110 (64-bit). Got a test project I can run to verify?

**Comment by Vanuan** [comment]
@bbottema How is it bad? Do you use babel or webpack for es modules?

I know that people sometimes think it's bad because they don't see imported symbols, but it's actually fine.

**Comment by bbottema** [comment]
@Vanuan, I only see transpiled code. Chrome reports sourcempas detected, but doesn't actually show them.

- Karma 1.7.0
- Webpack 2.6.1 with awesome-typesscript-loader 3.1.2
- devtool: 'source-map' in webpack-config, not using Uglify plugin
- "sourceMap": true in ts-config (typescript 2.3.1)

I don't know what to do next to make this work

**Comment by Vanuan** [comment]
Ah, so you're not using babel? You should probably awesome-typescript-loader authors.

**Comment by Coridyn** [comment]
For people serving their JS assets with Sprockets or the Rails Asset Pipeline be aware that it adds an extra newline at the top of processed sources which can cause debugging with sourcemaps to be off-by-one.

I'm adding this comment here in case people have a similar setup (using Webpack to build assets but serving files with Sprockets) to let them know the issue isn't with Webpack but could be with Sprockets/Asset Pipeline.

This issue was fixed in [this commit](https://github.com/rails/sprockets/commit/6012c9223f8289bb787e7d1accb3184ebe70ee10) which will be included in Sprockets v4.0.0.

**Comment by wbern** [comment]
webpack 3.6.0-3.7.1 using `eval-source-map` in Chrome does not let me place breakpoints inside jsx.

Works using regular "source-map" option.

**Comment by reohjs** [comment]
Something seems to have changed in 4.0.1 - With `devtool: 'source-map'`, `target: node` and the `source-map-support` package, instead of a bundled code paths I'm now getting stack trace paths like: 

```
C:\project\dist\webpack:\index.js:3:1
```
where the bundle.js is in `dist` and `index.js` is a source file.

`eval-*` options are now working for me too, though errors inside async/await code show generated code.

**Comment by reohjs** [comment]
I'm not using the loader, but thanks for the suggestion.

**Comment by gkatsanos** [comment]
https://github.com/webpack/webpack/issues

**Comment by Harrix** [comment]
Only the source-map and source-map and inline-source-map parameters work correctly with webpack 4.10.2 in Chrome  67.
![source-map](https://user-images.githubusercontent.com/2528139/40934211-535dc83a-685e-11e8-84cd-d84de6e121e9.png)
![eval-source-map](https://user-images.githubusercontent.com/2528139/40934212-5391b6fe-685e-11e8-8c82-fab53342e34b.png)

**Comment by MLoughry** [comment]
I don't know a lot about source-maps, but from our experience it seems that `cheap-module-eval-source-map` and `cheap-module-source-map` work well, but `source-map` has issues. This is confirmed by sourcemaps.io:

Mousetrap chunk with `source-map`: https://sourcemaps.io/report/1530635355948_https%3A%2F%2Fow1.res.office365.com%2Fowamail%2F20180702006%2Fscripts%2Fowa.21.js
Mousetrap chunk with 'cheap-module-source-map`: https://sourcemaps.io/report/1530635141846_https%3A%2F%2Fowamaildev.blob.core.windows.net%2Fowamail-branch%2Fcheap-module-source-map%2Fscripts%2Fowa.2.js

Briefly looking through the code, it seems that the `columns` property to `SourceMapDevToolPlugin` is the only difference. However, I don't claim to understand how that difference works in the only code that references it: https://github.com/webpack/webpack-sources/blob/91862553dc108e897ec24ece7b5bd8aae42620e5/lib/OriginalSource.js#L60

**Comment by MLoughry** [comment]
Sorry, that last comment was using webpack 4.12.0, for clarification.

**Comment by Tiberriver256** [comment]
Here's one that's a few lines off. Seemed to show up when I switched to non-relative imports. The link to sokra's site was a little too long for Github so I put it out on pastebin.

https://pastebin.com/xBDX2Vym

**Comment by lee149205** [comment]
i use devtool: 'source-map'
It's ok . But I cannot watch variable (always "not available") 
<img width="1182" alt="screen shot 2018-08-22 at 8 46 11 am" src="https://user-images.githubusercontent.com/11714006/44438142-f9b1af00-a5e7-11e8-8bbd-ef2f70d83dbf.png">

**Comment by lee149205** [comment]
oh, nevermind
I try comment uglify, it works well with devtool: 'source-map'
 plugins: [
            //uglify
 ]

**Comment by FabianPiconeDev** [comment]
Chrome 68.0.3440.106

Since a few days, sometimes on specific lines, i can't set a breakpoint. Since 10 months i am using `eval-source-map`, never had problems before.

Tried with `source-map` and `cheap-module-eval-source-map` and `cheap-eval-source-map`. Does not change anything.

**Comment by jxintang** [comment]
@FabianPiconeDev 
Having this same issue. updating chrome to 69 solves my problem.

**Comment by russelldavis** [comment]
`cheap-module-source-map` causes breakpoints to stop working in Chrome. See https://github.com/facebook/create-react-app/issues/6074#issuecomment-462087174.

**Comment by nsaikiran** [comment]
Hi everyone,
I'm using Chrome `Version 72.0.3626.121 (Official Build) (64-bit)`.
and Webpack `4.4.22`.
Webpack configuration is according to production guide from documentation.
Chrome says it detected sourcemaps. I can access original source code. **But while debugging can't inspect variables.**
`development` build is working fine with sourcemaps.
Any help is appreciated.
Thanks

**Comment by Akimyou** [comment]
> Hi everyone,
> I'm using Chrome `Version 72.0.3626.121 (Official Build) (64-bit)`.
> and Webpack `4.4.22`.
> Webpack configuration is according to production guide from documentation.
> Used `devtool`: source-map.
> Chrome says it detected sourcemaps. I can access original source code. **But while debugging can't inspect variables.**
> `development` build is working fine with sourcemaps.
> Any help is appreciated.
> Thanks

I have the same issue.

**Comment by kane-mason** [comment]
Hi, not sure i am in the right place. But i have always had issues with async/await being transformed to regeneratorRuntime. It is impossible to step through the code, you get sent into the source code. We work around this by just not transpiling async/await during development, but would prefer a proper solution. 

We have a simple issue with private fields, babel allows us to transpile them, but they are not inspectable in Chrome :(

**Comment by testacode** [comment]
Sourcemaps not working for me on webpack 5.6.0 no matter which type I use. I always get the transpiled code for development mode, I need to get the original code.

**Comment by evilebottnawi** [comment]
@testacode when you write it, please provide reproducible test repo, how we can help with this information?

**Comment by alexander-akait** [comment]
I want to close the problem due to a large number of reports that we cannot check, i.e. `same here`, `don't work` and etc without examples/repos/screenshots/etc. If you have the problem with source maps please open a new issue with steps how we can reproduce it. Also many issues were solved in webpack@5, so I recommend to check this before. Thanks

**Comment by ColCh** [comment]
@alexander-akait there is a link for this issue in docs

https://webpack.js.org/configuration/devtool/#devtool:~:text=There%20are%20some%20issues%20with%20Source%20Maps%20in%20Chrome

May be remove that reference as well? Opening closed issue using that link would provide WTF signal from people :)

**Comment by alexander-akait** [comment]
@ColCh Thanks, https://github.com/webpack/webpack.js.org/issues/4606, we will remove it
